controllers/comments.controllers.js

```diff
@@ -14,4 +19,27 @@ exports.getCommentsByArticleId = ( req , res , next ) => {
     .send( { comments } ) ;
   })
   .catch( next ) ;
+} ;
+
+exports.postCommentByArticleId = ( req , res , next ) => {
+  const { article_id } = req.params ;
+  const { username , body } = req.body ;
+
+  if ( !username || !body ) {
+    throw( { status : 400 , msg : "Bad request" } );
+  }
+
+  Promise.all([
```

Comment:
This is a lovely function! However, article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value

Reply:
This means you don't need to check separately (comment these checks out and run the code to see what I mean) - you have psql error handlers which should handle this for you